### PR TITLE
Fix bulk delete when entity has empty components

### PR DESCRIPTION
**Pull request by Marc-Roig**
### What does it do?

Fix https://github.com/strapi/strapi/issues/14384

### Why is it needed?

Deleting an entity with empty components was throwing an error.
`deleteComponents` function expects one of the two:
- An entity with loaded components.
- An entity with an id.

When components were empty, the first case was not met, so the function was expecting the entity with an id, which we were not passing. So 💥 

### How to test it?

Delete or bulk delete an entity and you should not see any error in the console.

### Related issue(s)/PR(s)

https://github.com/strapi/strapi/issues/14384

### CODE REVIEW

File: packages/core/strapi/lib/services/entity-service/index.js [hunk 1]
```diff
@@ -217,7 +217,7 @@ const createDefaultImplementation = ({ strapi, db, eventHub, entityValidator })
     const componentsToDelete = await getComponents(uid, entityToDelete);
 
     await db.query(uid).delete({ where: { id: entityToDelete.id } });
-    await deleteComponents(uid, { ...entityToDelete, ...componentsToDelete });
+    await deleteComponents(uid, componentsToDelete);
```
Comment (petersg83) [inline packages/core/strapi/lib/services/entity-service/index.js:220]: In entityToDelete the id was not there?

Comment (Marc-Roig) [reply packages/core/strapi/lib/services/entity-service/index.js:220]: yes! But it wasn't for the bulk delete.

I though it was going to complicate things doing this in the bulkDelete, when you have multiple entities. 
So my proposal is to return the entity id in `getComponents` instead

Comment (Marc-Roig) [reply packages/core/strapi/lib/services/entity-service/index.js]: After discussing it with you I made the changes!
It didn't make sense to return the entity ID in `getComponents`. Now we specify if we want to loadComponentss when calling `deleteComponents` 👍

**Review by jhoward1994 — CHANGES_REQUESTED** [review]
Nice!

In my testing single deletion works but `bulkDelete` is causing an error. I think it is caused by `packages/core/strapi/lib/services/entity-service/components.js#L361` not passing the third argument to `deleteComponents`

p.s. do you know how to leave a PR comment on an unmodified line?

**Comment by Marc-Roig** [comment]
> Nice!
> 
> In my testing single deletion works but `bulkDelete` is causing an error. I think it is caused by `packages/core/strapi/lib/services/entity-service/components.js#L361` not passing the third argument to `deleteComponents`
> 
> p.s. do you know how to leave a PR comment on an unmodified line?

good call. 
I had to add a default {} value for for the `loadComponents` [parameter](https://github.com/strapi/strapi/pull/14600/commits/69fa150e59e379d56ef7d3ef8e5f85e4884d9662)

In the `deleteComponents` function the desired behaviour is to load the components (which is the default when not passing anything)

And no idea how to leave a comment on an unmodified line :/

**Comment by jhoward1994** [comment]
> > Nice!
> > In my testing single deletion works but `bulkDelete` is causing an error. I think it is caused by `packages/core/strapi/lib/services/entity-service/components.js#L361` not passing the third argument to `deleteComponents`
> > p.s. do you know how to leave a PR comment on an unmodified line?
> 
> good call. I had to add a default {} value for for the `loadComponents` [parameter](https://github.com/strapi/strapi/pull/14600/commits/69fa150e59e379d56ef7d3ef8e5f85e4884d9662)
> 
> In the `deleteComponents` function the desired behaviour is to load the components (which is the default when not passing anything)
> 
> And no idea how to leave a comment on an unmodified line :/

Nice, it works well for me now 🙂

**Review by jhoward1994 — APPROVED** [review]

**Review by petersg83 — CHANGES_REQUESTED** [review]

File: packages/core/strapi/lib/services/entity-service/components.js [hunk 2]
```diff
@@ -274,44 +274,38 @@ const deleteOldDZComponents = async (uid, entityToUpdate, attributeName, dynamic
...
-      continue;
-    }
-
-    if (attribute.type === 'dynamiczone') {
-      const value =
-        entityToDelete[attributeName] ||
-        (await strapi.query(uid).load(entityToDelete, attributeName));
+      if (!value && loadComponents) {
+        value = await strapi.query(uid).load(entityToDelete, attributeName);
+      }
```
[the remark this answers is not in the document]
Comment (Marc-Roig) [reply packages/core/strapi/lib/services/entity-service/components.js]: you are right. This could be removed.

**Comment by dkryvi** [comment]
Hello, any updates in this pull request? It's a critical issue for our project

**Review by petersg83 — APPROVED** [review]
LGTM :) well done

**Pull request merged**